eucaconsole/views/instances.py

```
@@ -165,6 +165,20 @@ def get_monitoring_state(self, instance=None):
         if self.cloud_type == 'aws':
             return _(u'Detailed') if instance.monitoring_state == 'enabled' else _(u'Basic')
 
+    def get_termination_protection_state(self, instance=None):
+        """
+        Get termination protection state as a tuple of state (bool) and label
+        :param instance: Boto EC2 instance object
+        :return: tuple of is_enabled (True or False) and label ('Enabled' or 'Disabled')
+        :rtype: tuple
+        """
+        if not instance:
```

Shouldn't the instance view fail fast if the instance can't be found? Throwing a 404 instead of None checks is cleaner.